kaldb/src/main/java/com/slack/kaldb/zipkinApi/ZipkinService.java

```diff
@@ -0,0 +1,591 @@
+package com.slack.kaldb.zipkinApi;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.linecorp.armeria.server.annotation.*;
+import com.slack.kaldb.elasticsearchApi.searchResponse.*;
+import com.slack.kaldb.logstore.LogMessage;
+import com.slack.kaldb.logstore.LogWireMessage;
+import com.slack.kaldb.proto.service.KaldbSearch;
+import com.slack.kaldb.server.KaldbQueryServiceBase;
+import com.slack.kaldb.util.JsonUtil;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+@SuppressWarnings(
+    "OptionalUsedAsFieldOrParameterType") // Per https://armeria.dev/docs/server-annotated-service/
+public class ZipkinService {
+  private final KaldbQueryServiceBase searcher;
+
+  public ZipkinService(KaldbQueryServiceBase searcher) {
+    this.searcher = searcher;
+  }
+
+  /**
+   * Get Services API
+   *
+   * @return a list of all service names associated with span endpoints.
+   * @see <a href="https://zipkin.io/zipkin-api/#/default/get_services">API doc</a>
+   */
+  @Get
+  @Path("/services")
+  public String getServices() throws IOException {
+    GetServiceResult results = new GetServiceResult(ImmutableList.of("s1", "s2"));
+    return JsonUtil.writeJsonArray(JsonUtil.writeAsString(results.getServices()));
+  }
+
+  /**
+   * Get Span Names API
+   *
+   * @return all the span names recorded by a particular service
+   * @see <a href="https://zipkin.io/zipkin-api/#/default/get_spans">API doc</a>
+   */
+  @Get("/spans")
+  public String getSpans(@Param("serviceName") Optional<String> serviceName) throws IOException {
```

Comment:
According to the documentation https://zipkin.io/zipkin-api/#/default/get_spans , `serviceName` is a required field. So I think we can remove the `Optional<String>` and make it just a `String` ?

Reply:
I think the next step here is to add start adding some of the additional parameters the GET spans api supports and using it when creating the `searchRequestBuilder`